frontend/app/routes/protected/person-case/primary-docs.tsx

```diff
@@ -179,39 +181,25 @@ interface CurrentStatusInCanadaProps {
   defaultValue?: string;
   errorMessage?: string;
   onChange?: React.ChangeEventHandler<HTMLSelectElement>;
+  statusInCanada: LocalizedApplicantStatusInCanadaChoice[];
 }
 
-function CurrentStatusInCanada({ defaultValue, errorMessage, onChange }: CurrentStatusInCanadaProps) {
+function CurrentStatusInCanada({ defaultValue, errorMessage, onChange, statusInCanada }: CurrentStatusInCanadaProps) {
   const { t } = useTranslation(handle.i18nNamespace);
-  const CurrentStatusInCanada = [
-    'canadian-citizen-born-in-canada',
-    'canadian-citizen-born-outside-canada',
-    'registered-indian-born-in-canada',
-    'registered-indian-born-outside-canada',
-    'permanent-resident',
-    'temporary-resident',
-    'no-legal-status-in-canada',
-  ] as const;
 
-  const currentStatusInCanadaOptions = [
-    {
-      children: t('protected:request-details.select-option'),
-      value: '',
-    },
-    ...CurrentStatusInCanada.map((value) => ({
-      value: value,
-      children: t(`protected:primary-identity-document.current-status-in-canada.options.${value}` as const),
-      disabled: value != 'canadian-citizen-born-outside-canada',
-    })),
-  ];
+  const currentStatusInCanadaOptions = [{ id: 'select-option', name: '' }, ...statusInCanada].map(({ id, name }) => ({
+    value: id === 'select-option' ? '' : id,
+    children: id === 'select-option' ? t('protected:request-details.select-option') : name,
+    disabled: id != '900000001',
```

Comment:
Use [Strict inequality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_inequality) (!==)
```suggestion
    disabled: id !== '900000001',
```